### DISCUSSION TITLE
Plotting the VIP

### DISCUSSION THREAD

**Discussion by MoMaz123** [kogalur/randomForestSRC#252] [Q&A] [answered]
Hi 
My model is running very well, but I cant  get the VIP plot, 

obj <- rfsrc(Surv(ihd_time, ihd_ep)~.,data_wide2,
             ntree = 100, nodesize = 5, nsplit = 10, importance = TRUE)
print(obj)
get.cindex(obj$yvar[,1], obj$yvar[,2], obj$predicted.oob)

jk.obj <- subsample(obj)
plot(jk.obj, xlab = "Variable Importance (x 100)")

Keep having this error, 

plot(jk.obj, xlab = "Variable Importance (x 100)")
Error in dimnames(x) <- dn : 
  length of 'dimnames' [1] not equal to array extent

Thanks for help,

**Comment by ishwaran** [comment, marked as answer]
You have to use `plot.subsample` with objects created by `subsample`
See the help file: https://www.randomforestsrc.org/reference/plot.subsample.rfsrc.html
Here's a worked example:
 
```
o <- rfsrc(Ozone ~ ., airquality)
oo <- subsample(o)
plot.subsample(oo)
```

If you just want VIMP (again see the help file `rfsrc` or `vimp`):
```
o <- rfsrc(Ozone ~ ., airquality, importance=TRUE)
plot(o)
```

**Reply by MoMaz123** [reply]
Great! many  thanks

**Comment by liborpastika** [comment]
Hi, is there a simple way of ordering the variables by the magnitude of their variable importances?

Thank you for your help

**Comment by ishwaran** [comment]
See the following example

```
o <- rfsrc(mpg~.,mtcars,importance="permute")
o$xvar.names[order(o$importance,decreasing=T)]

1] "disp" "wt"   "cyl"  "hp"   "drat" "carb" "vs"   "gear" "am"   "qsec"
```
